lib/export/coldp/files/type_material.rb

```diff
@@ -0,0 +1,153 @@
+# ID
+# nameID
+# citation
+# status
+# referenceID
+# locality
+# country
+# latitude
+# longitude
+# altitude
+# host
+# date
+# collector
+# institutionCode
+# catalogNumber
+# associatedSequences
+# sex
+# link
+# remarks
+#
+module Export::Coldp::Files::TypeMaterial
+
+  def self.locality(co)
+    [co.collecting_event&.cached_level0_geographic_name, co.collecting_event&.cached_level1_geographic_name, co.collecting_event&.cached_level2_geographic_name].compact.join(', ')
+  end
+
+  # TODO: This very likely includes more than just hosts (symbiotypes):
+  #   could possibly be narrowed down by checking if the biological relationship type's inverted name includes the word host?
+  #     we are specifying the relationship type in the string to make it clearer
+  def self.host(co)
+    if co.biological_associations == []
+      return nil
+    end
+    hosts = []
+    co.biological_associations.each do |ba|
+      if ba.biological_association_object_type == 'Otu'
+        o = Otu.find(ba.biological_association_object_id)
+        if o.taxon_name_id.nil?
+          object_name = o.name
+        else
+          tn = TaxonName.find(Otu.find(ba.biological_association_object_id).taxon_name_id)
+          object_name = tn.cached
+        end
+        rel_type = BiologicalRelationship.find(ba.biological_relationship_id)
+        if rel_type.inverted_name.include? 'host'
+          hosts.append("#{rel_type.name} #{object_name}")
+        end
+      end
+    end
+    hosts.to_sentence
+  end
+
+  # CollectingEvent.verbatim_date seems to be in different formats (e.g., "6.VIII.2018", "14 August 1992")
+  # so preferably take start_date_xxxx and format to ISO 8601
+  def self.date(co)
+    date = [co.collecting_event&.start_date_year, co.collecting_event&.start_date_month, co.collecting_event&.start_date_day].compact.join('-')
+    if co.collecting_event&.start_date_year.nil? or date.nil?  # if year is nil, verbatim_date might be better?
+      date = co.collecting_event&.verbatim_date
+    end
+    date
+  end
+
+  def self.collector(co)
```

Comment:
You should use the DwC methods likely, see the inlucde for CollectionObjects.  Something like co.dwc_<field_in_snake_case>.  Many, many options there.